### PR TITLE
Use latest props when updating value cache (fixes issue #59)

### PR DESCRIPTION
**Pull request by gitname**
Prior to this PR, this component processed `nextProps.values` using props that belonged to a different props set (e.g. using `props.endDate` instead of `nextProps.endDate`). As a result, the behavior described in Issue #59 would occur.

Using the code in this PR, the component always uses the props that belong to the same props set as the `values` prop it is processing. In other words, the component uses `props.*` when processing `props.values`, and uses `nextProps.*` when processing `nextProps.values`. That fixes Issue #59.

This PR includes a test case that tests for Issue #59.

### CODE REVIEW

File: src/index.jsx [hunk 1]
```diff
@@ -11,45 +11,49 @@ class CalendarHeatmap extends React.Component {
...
-    const { startDate, endDate, numDays } = this.props;
+    const { startDate, endDate, numDays } = this.latestProps;
     if (numDays) {
       // eslint-disable-next-line no-console
       console.warn('numDays is the deprecated prop, so it will be removed in the next release. Consider using of startDate prop instead.');
       return numDays;
     }
     const timeDiff = endDate - startDate;
-    return Math.ceil(timeDiff / (1000 * 3600 * 24));
+    return Math.ceil(timeDiff / MILLISECONDS_IN_ONE_DAY);
```
Comment (gitname) [inline src/index.jsx]: I updated the `getDateDifferenceInDays` method to use the `MILLISECONDS_IN_ONE_DAY` constant instead of calculating the same value itself. This change is unrelated to the rest of this PR.

**Comment by okize** [comment]
thanks for all your efforts here @gitname! this is an important bugfix and improvement

this has been published as v1.6.2
